db/webscraper.py

```diff
@@ -0,0 +1,264 @@
+    return object_attributes
+
+# Processes a list of object attributes and returns a list of the format:
+#       [
+#           [ crn,
+#             department,
+#             name,
+#             url ]
+#       ]
+def get_urls(object_attributes):
+    course_urls = []
+
+    for course in object_attributes:
+        output = re.search(pattern, unicode(course)) # match regex to link
+
+        # access using:
+        catoid = output.group('catoid')
+        coid = output.group('coid')
+        display_options = output.group('display_options')
+        crn = output.group('crn')
+        name = output.group('name')
+
+        # Request url based on the regex groups catoid, coid and display_options
+        # Format: http://catalog.swarthmore.edu/ajax/preview_course.php?catoid=7&coid=7746&display_options=a:2:{s:8:~location~;s:7:~program~;s:4:~core~;s:4:~4085~;}&show
+        course_url = 'http://catalog.swarthmore.edu/ajax/preview_course.php?catoid={}&coid={}&display_options={}&show'.format(catoid, coid, display_options)
+
+        # Separate the CRN from the Department
+        # TODO: Check this works
+        dept = crn.split()[0]
+        crn = crn.split()[1]
+
+        # Add course info and url the list of urls
+        course_urls.append([crn, name, dept, course_url])
+
+    return course_urls
+
+# Scrapes a single course page for all relevant information and returns course
+# data in a dictionary that can be easily converted to json.
+# Note: course_url is of the same format as that of the get_urls() function.
+def scrape(course_url):
+    # Set up the course object with known values
+    course = {}
+    course['crn'] = course_url[0]
+    course['name'] = course_url[1][1:]
+    course['dept'] = course_url[2]
+    if 'First-Year Seminar:' in course_url[2]:
+        course['FYS'] = True
+    else:
+        course['FYS'] = False
+
+    # Scrape!
+    course_response = requests.get(course_url[3])
+    course_soup = BeautifulSoup(course_response.content, 'lxml')
+    course_tag = course_soup.find('div', class_="ajaxcourseindentfix", style=False)
+    strings = course_tag.stripped_strings
+
+    # In progress: finding descriptions
+    '''# Get description
+    description = ""
+    flip = True
+    start = course_soup.find_all('hr')[0]
+    end = start.find_next('br')
+    current = start.next_sibling
+    # If the course does not have a description
+    if not isinstance(current, NavigableString):
+        description = 'N/A'
+    else:
+        while current != end:
+            if current.name == 'a':
+                description += current.string
+            elif current.name == 'p':
+                break
+            elif current.name == 'span':
+                pass
+            elif current.name == 'h5':
+                pass
+            else:
+                description += unicode(current)
+            current = current.next_sibling
+
+    course['description'] = description'''
```

Comment:
have you played with .contents? does this help? http://stackoverflow.com/questions/16835449/python-beautifulsoup-extract-text-between-element